test/assignPermissionToRole.spec.js

```diff
@@ -1,77 +1,59 @@
-                "resource_type" : "db",
-                "resource" : ["nodeDB"]
-            };
-
-            conn.assignPermissionToRole({ role: "myrole", permissionObj: aNewPermission }, function (data, response) {
-                expect(response.statusCode).to.be(404);
-                done();
-            });
-        });
-
-        it("should pass assinging a Permissions to a new role.", function (done) {
-
-            var aNewRole = "newtestrole";
-            var aNewPermission = {
-                "action" : "write",
-                "resource_type" : "db",
-                "resource" : ["nodeDB"]
-            };
-
-            conn.deleteRole({ role: aNewRole}, function() {
-                // delete role if exists
-
-                conn.createRole({ rolename: aNewRole }, function (data, response1) {
-                    expect(response1.statusCode).to.be(201);
-
-                    conn.assignPermissionToRole({ role: aNewRole, permissionObj: aNewPermission }, function (data, response2) {
-
-                        expect(response2.statusCode).to.be(201);
-
-                        // delete role
-                        conn.deleteRole({ role: aNewRole }, function (data, response3) {
-                            expect(response3.statusCode).to.be(200);
-                            done();
-                        });
-                    });
-                });
-            });
-        });
+const RandomString = require('randomstring');
+const Stardog = require('../lib');
+const {
+  seedDatabase,
+  dropDatabase,
+  generateDatabaseName,
+  generateRandomString,
+} = require('./setup-database');
+
+describe('assignPermissionToRole()', () => {
+  const database = generateDatabaseName();
+  var conn;
+
+  beforeAll(seedDatabase(database));
+  afterAll(dropDatabase(database));
+
+  beforeEach(() => {
+    conn = new Stardog.Connection();
+    conn.setEndpoint('http://localhost:5820/');
+    conn.setCredentials('admin', 'admin');
+  });
+
+  it('should fail trying to assign a permssion to a non-existent role.', done => {
+    var aNewPermission = {
+      action: 'write',
+      resource_type: 'db',
+      resource: [database],
+    };
+
+    conn.assignPermissionToRole(
+      { role: 'myrole', permissionObj: aNewPermission },
+      (data, response) => {
+        expect(response.statusCode).toEqual(404);
+        done();
+      }
+    );
+  });
+
+  it('should pass assinging a Permissions to a new role.', done => {
+    var aNewRole = generateRandomString();
+    var aNewPermission = {
+      action: 'write',
+      resource_type: 'db',
+      resource: [database],
+    };
+
+    conn.createRole({ rolename: aNewRole }, (data, response1) => {
+      expect(response1.statusCode).toEqual(201);
+
+      conn.assignPermissionToRole(
+        { role: aNewRole, permissionObj: aNewPermission },
+        (data, response2) => {
+          expect(response2.statusCode).toEqual(201);
+          done();
+        }
+      );
```

Comment:
I dropped all that cleanup code because we destroy the entire database after each spec is complete. Makes the tests much more manageable.